guides/k8-akka-cluster/src/main/paradox/index.md

```diff
@@ -0,0 +1,600 @@
+
+Please note that you must choose to enable either `JavaAppPackaging` or `JavaServerAppPackaging` - you _can't_ enable both.
+
+We will now configure the Docker settings required to containerize the application.
+
+Append the Akka clustering related sytem properties to the `dockerEntrypoint` setting within the module settings:
+
+```
+lazy val myApp = project("my-app")
+  .settings(
+    // Add the following line within the module settings
+    dockerEntrypoint ++= Seq(
+      """-Dakka.remote.netty.tcp.hostname="$AKKA_REMOTING_BIND_HOST"""",
+      """-Dakka.remote.netty.tcp.port="$AKKA_REMOTING_BIND_PORT"""",
+      """-Dakka.cluster.seed-nodes.0="akka.tcp://${AKKA_ACTOR_SYSTEM_NAME}@${AKKA_SEED_NODE_HOST}:${AKKA_SEED_NODE_PORT}"""",
+      "-Dakka.io.dns.resolver=async-dns",
+      "-Dakka.io.dns.async-dns.resolve-srv=true",
+      "-Dakka.io.dns.async-dns.resolv-conf=on"
+    )
+```
+
+As part of building the Docker image, SBT Native Packager will provide its own Docker entry point script to start the application which accepts additional arguments. When system properties are presented as part of the arguments, they will be appended to the JVM options when the application is started within the container.
+
+Next we will transform the generated Dockerfile `ENTRYPOINT` instruction:
+
+```
+lazy val myApp = project("my-app")
+  .settings(
+    // Add the following line within the module settings
+    dockerCommands :=
+      dockerCommands.value.flatMap {
+        case ExecCmd("ENTRYPOINT", args @ _*) => Seq(Cmd("ENTRYPOINT", args.mkString(" ")))
+        case v => Seq(v)
+      }
+   )
+```
+
+As part of the Docker build, SBT Native Packager will also generate a startup script for the application. This startup script is referenced by the `ENTRYPOINT` command in the Dockerfile, i.e. `ENTRYPOINT ["bin/myapp"]`. Note the value of the `ENTRYPOINT` is an array - this is the `exec` form declaration of the `ENTRYPOINT`.
+
+The script above transforms `ENTRYPOINT` from `exec` form to `shell` form. In the `shell` form, the `ENTRYPOINT` is declared as a simple string value. This string value will be executed by the container's shell using `sh -c`. The `shell` form is required to ensure the environment variables declared in the `dockerEntrypoint` argument is sourced from the shell within the Docker container. Refer to Docker [ENTRYPOINT](https://docs.docker.com/engine/reference/builder/#entrypoint) documentation for the difference between `exec` and `shell` form.
+
+The container image will be published to the the default Docker repository specified by SBT Native Packager. To publish to a different repository, set the `dockerRepository` setting to the repository you wish to publish. Please note this is an optional step. The Kubernetes deployment in this guide expects the image to be located at `mygroup/myapp`. To match the repository we need to set the `dockerRepository` to `mygroup`:
+
```

Comment:
These optional setting get confusing because they are optional for the end user, but if they are following this example, they are required, right? I would list the settings in the introduction, and explain which are optional. Then, I'd explain that in this exercise we will use the optional settings. Then, you can remove the "optional" explanation from the steps themselves.

Remove last sentence from previous paragraph, replace with:

5. Add the following line in the `.settings` section to set the `dockerRepository` to `mygroup`: `dockerRepository := Some("mygroup")`.   The resulting file will look similar to the following: